File: Candlelight.Core/Dtos/Game/GameDetailsDto.cs
using Candlelight.Core.Entities.Steam;

namespace Candlelight.Core.Dtos.Game;
public class GameDetailsDto
{
    public Guid Id { get; set; }
    public int? AppId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? DescriptionSnippet { get; set; }
    public string? Description { get; set; }
    public string? HeaderImage { get; set; }
    public string? Developer { get; set; }
    public string? Publisher { get; set; }
    public bool? IsFree { get; set; }
    public decimal? Price { get; set; }
    public string? Currency { get; set; }
    public int? MetacriticScore { get; set; }
    public string? Website { get; set; }
    public DateTime? ReleaseDate { get; set; }
    public int ModCount { get; set; }
    public int FavouriteCount { get; set; }
    public bool IsCustom { get; set; }

    public List<Genre> Genres { get; set; } = [];
    public List<Category> Categories { get; set; } = [];
    public List<Platform> Platforms { get; set; } = [];
}

File: candlelight.client/src/app/components/pages/games/models/game-details-dto.ts
export interface Genre {
  id: string;
  name: string;
}

export interface Category {
  id: string;
  name: string;
}

export interface Platform {
  id: string;
  name: string;
}

export interface GameDetailsDto {
  id: string;
  appId: number | null;
  name: string;
  descriptionSnippet?: string;
  description?: string;
  headerImage?: string;
  developer?: string;
  publisher?: string;
  isFree?: boolean;
  price?: number;
  currency?: string;
  metacriticScore?: number;
  website?: string;
  releaseDate?: string;
  modCount: number;
  favouriteCount: number;
  isCustom: boolean;
  genres: Genre[];
  categories: Category[];
  platforms: Platform[];
}
